src/terrain_builder.cpp

```diff
@@ -1176,6 +1176,34 @@ void terrain_builder::build_terrains()
 		}
 
 	}
+
+	for(int x = -2; x <= map().w(); ++x) {
+		for(int y = -2; y <= map().h(); ++y) {
+			const map_location loc(x,y);
+
+			const imagelist* img_list_fg = get_terrain_at(loc, "", FOREGROUND);
+			if (img_list_fg != NULL) {
+				BOOST_FOREACH(const animated<image::locator>& locator, *img_list_fg){
+					for(size_t i=0; i<=locator.get_frames_count(); i++) {
```

Comment:
Prefer using prefix increment/decrement instead of postfix (i.e. `++i` as opposed to `i++`) wherever you don't actually need postfix. Also, the spacing around `=` and `<=` operators here and on line 1198 could be better.

Also, for future reference, you probably want std::cerr instead of std::cout, since the latter is buffered and may come out of order relative to cerr (or never at all, in the event of an abnormal exit/crash).